test/test_reductions.py

```diff
@@ -53,8 +57,142 @@ def _rand_shape(dim, min_size, max_size):
         shape.append(random.randint(min_size, max_size))
     return tuple(shape)
 
+
 class TestReductions(TestCase):
 
+    ###########################################################################
+    # ReductionOpInfo unit tests
+    ###########################################################################
+
+    @ops(reduction_ops)
```

Comment:
Would be great to have a way to add another decorator to parametrize a test with custom parameters. Or specify extra parameters to use in the @ops decorator?

Reply:
Elaborating, in this case the decorator would be used to add a keepdim boolean parameter to each of the dim tests. While the code inside the tests could just iterate over [True, False] to test both cases, there are many operators that don't accept certain combinations of dim and keepdim and would end up failing most dim related tests this way. But by parametrizing and having separate tests for these cases, operators can then skip only the cases where they fail and that also gives more visibility to the problems that need to be fixed.

Reply:
@jbschlosser's got a PR adding a pytest-like paramtrize decorator that should do this